prow/tide/tide.go

```diff
@@ -822,11 +822,19 @@ func (c *Controller) mergePRs(sp subpool, prs []PullRequest) error {
 	for i, pr := range prs {
 		log := log.WithFields(pr.logFields())
 		mergeMethod := c.config().Tide.MergeMethod(sp.org, sp.repo)
-		if squashLabel := c.config().Tide.SquashLabel; squashLabel != "" {
+		squashLabel := c.config().Tide.SquashLabel
+		rebaseLabel := c.config().Tide.RebaseLabel
+		mergeLabel := c.config().Tide.MergeLabel
+		if squashLabel != "" || rebaseLabel != "" || mergeLabel != "" {
 			for _, prlabel := range pr.Labels.Nodes {
 				if string(prlabel.Name) == squashLabel {
 					mergeMethod = github.MergeSquash
 					break
+				} else if string(prlabel.Name) == rebaseLabel {
+					mergeMethod = github.MergeRebase
+					break
+				} else if string(prlabel.Name) == mergeLabel {
+					mergeMethod = github.MergeMerge
```

Comment:
Don't we want to break here too?

Reply:
I took the breaks out, since if we want to check for multiple labels, we need to always loop through all the labels.

Reply:
Could still break when setting `MergeInvalid`; no need to keep looping through labels to check if it's doubly-invalid. (Or flip the logic around; start out with a nil `mergeMethod`, loop looking for a label specifying one & set it. Continue looping, and if you get another explicit label, set to `MergeInvalid` and pull the cord. Outside the loop, assign to `c.config().Tide.MergeMethod(sp.org, sp.repo)` if nil. It's a beautifully hairy yak.)